cmd/metal-api/internal/service/network-service.go

```diff
@@ -439,6 +456,79 @@ func validateAdditionalAnnouncableCIDRs(additionalCidrs []string, privateSuper b
 	return nil
 }
 
+func validatePrefixesAndAddressFamilies(prefixes, destinationPrefixes []string, defaultChildPrefixLength metal.ChildPrefixLength, privateSuper bool) (metal.Prefixes, metal.Prefixes, metal.AddressFamilies, error) {
+
+	pfxs, addressFamilies, err := validatePrefixes(prefixes)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	// all DestinationPrefixes must be valid and from the same addressfamily
+	destPfxs, destinationAddressFamilies, err := validatePrefixes(destinationPrefixes)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	if len(destinationAddressFamilies) > len(addressFamilies) {
```

Comment:
Maybe this check is insufficient, e.g. there are IPv6 destination prefixes but only IPv4 prefixes.

Reply:
done